File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int element = nums[0];
        int num = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == element) {
                num++;
            } else {
                num--;
                if (num == 0) {
                    element = nums[i];
                    num = 1;
                }
            }
        }

        return element;
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        count_map = {}

        for num in nums:
            count_map[num] = count_map.get(num, 0) + 1
        
        major_elements = []
        number = int(len(nums) / 3)
        for key, value in count_map.items():
            if value > number:
                major_elements.append(key)

        return major_elements